### ISSUE TITLE
strange results with Faster RCNN and grayscale input 

### ISSUE THREAD

**Issue by adeschemps** [PyTorchLightning/lightning-bolts#770]
## 🐛 Bug

This is a copy-paste of this issue (https://github.com/PyTorchLightning/pytorch-lightning/discussions/10338) here, as it may be more appropriate on this page. 

The following code doesn't fail, even though I believe it should: 

```python
from pl_bolts.models.detection import FasterRCNN
import torch 

model = FasterRCNN()
print(model)
image = torch.randn((1,1,256,256))
result = model(image)
```

as `print(model)` returns: 

```
self.model: FasterRCNN(
  (transform): GeneralizedRCNNTransform(
      Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
      Resize(min_size=(800,), max_size=1333, mode='bilinear')
  )
  (backbone): BackboneWithFPN(
    (body): IntermediateLayerGetter(
      (conv1): Conv2d(3, 64, kernel_size=(7, 7), stride=(2, 2), padding=(3, 3), bias=False)
      (bn1): FrozenBatchNorm2d(64, eps=1e-05)
      (relu): ReLU(inplace=True)
      (maxpool): MaxPool2d(kernel_size=3, stride=2, padding=1, dilation=1, ceil_mode=False)
      (layer1): Sequential(
        (0): Bottleneck(
          (conv1): Conv2d(64, 64, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (bn1): FrozenBatchNorm2d(64, eps=1e-05)
          (conv2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
          (bn2): FrozenBatchNorm2d(64, eps=1e-05)
          (conv3): Conv2d(64, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (bn3): FrozenBatchNorm2d(256, eps=1e-05)
          (relu): ReLU(inplace=True)
          (downsample): Sequential(
            (0): Conv2d(64, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
            (1): FrozenBatchNorm2d(256, eps=1e-05)
          )
        )

```

showing that the first convolution of the backbone expects an input with 3 channels instead of one. This is confirmed by https://github.com/pytorch/vision/blob/3300692c6e7c2023d2f2356a69ec22ca91e38790/torchvision/models/resnet.py#L323: 

``` python
self.conv1 = nn.Conv2d(3, self.inplanes, kernel_size=7, stride=2, padding=3, bias=False)
``` 
### Expected behavior

I expected to receive some kind of error like `given weight tensor of size [blabla], expected input to have 3 channels but got input with 1 channel instead`. This isn't a massive issue, but I don't think this is intended behaviour.

**Comment by adeschemps** [comment]
After some investigation, this is related to the normalization code used in the GeneralizedRCNNTransform (https://github.com/pytorch/vision/blob/main/torchvision/models/detection/transform.py). 

```
    def normalize(self, image: Tensor) -> Tensor:
        if not image.is_floating_point():
            raise TypeError(
                f"Expected input images to be of floating type (in range [0, 1]), "
                f"but found type {image.dtype} instead"
            )
        dtype, device = image.dtype, image.device
        mean = torch.as_tensor(self.image_mean, dtype=dtype, device=device)
        std = torch.as_tensor(self.image_std, dtype=dtype, device=device)
        return (image - mean[:, None, None]) / std[:, None, None] 
```
`mean[:, None, None]` returns a `[nchannels, 1, 1]`-sized tensor, which then enables broadcasting with inconsistent number of channels. The same normalization with `torchvision.transforms.Normalize` fails, as expected.
